pkg/component/runtime/command.go

```diff
@@ -392,8 +404,16 @@ func (c *CommandRuntime) handleProc(state *os.ProcessState) bool {
 	switch c.actionState {
 	case actionStart:
 		// should still be running
-		stopMsg := fmt.Sprintf("Failed: pid '%d' exited with code '%d'", state.Pid(), state.ExitCode())
-		c.forceCompState(client.UnitStateFailed, stopMsg)
+		reportFailure := true
+		if c.restartBucket != nil {
+			// report failure only if bucket is full of restart events
+			reportFailure = tryAddToBucket(c.restartBucket)
```

Comment:
I think we may want to log when we are suppressing the failed state as well so we can observe this working properly.

Reply:
This would also let us notice when a restart we should not have suppressed gets suppressed.